src/main/clojure/clara/rules/dsl.clj

```diff
@@ -277,7 +277,7 @@
 
          query {:lhs (list 'quote (mapv #(resolve-vars % (destructure-syms %))
                                         conditions))
-                :params (set params)}
+                :params (set (map keyword params))}
```

Comment:
I don't mind allowing symbols as query params, but I suggest that it be restricted to keywords or symbols, not anything that keyword can coerce e.g. a string or something else that it will return nil on and cause problems later.  We probably should have a test validating that this fails as well.

Reply:
I've addressed this and also added corresponding tests to ensure the new coerce function works as intended, including a failure test.